pkg/pipeline/execution/logs/logs.go

```diff
@@ -0,0 +1,143 @@
+		name:    pipelineExecutionName,
+	}
+	rest := opts.RESTClientGetter()
+	c, err := client.NewClient(rest)
+	if err != nil {
+		return nil, err
+	}
+	pList.Client = c
+	return pList, nil
+}
+
+// LogsExecute fetches and displays aggregated logs from pipeline execution.
+func (p *pipelineLogs) LogsExecute() error {
+	namespace := p.args.Namespace
+
+	pipelineRun := &tektonapi.PipelineRun{}
+
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      p.name,
+	}
+
+	// Retrieve the PipelineRun object
+	if err := p.CtrlRuntimeClient().Get(context.Background(), namespacedName, pipelineRun); err != nil {
+		logrus.Errorf("failed to get PipelineRun '%s' in namespace '%s', %v", p.name, namespace, err)
+		return err
+	}
+
+	// Retrieve the ChildReferences of TaskRun
+	taskPodList := pipelineRun.Status.ChildReferences
+
+	// Iterate through each TaskRun to fetch their logs
+	for _, taskRef := range taskPodList {
+		if taskRef.Kind == "TaskRun" {
+			logrus.Infof("Fetching logs for TaskRun: %s", taskRef.Name)
+			if err := p.fetchAndPrintPodLogs(taskRef.Name, namespace); err != nil {
+				logrus.Errorf("failed to fetch logs for TaskRun '%s' in namespace '%s', %v", taskRef.Name, namespace, err)
+				// here just continue to attempt fetching logs for other TaskRuns, rather than directly return an error.
+			}
+		}
+	}
+
+	return nil
+}
+
+// fetchAndPrintPodLogs fetches and prints the logs of the pod associated with a given TaskRun.
+func (p *pipelineLogs) fetchAndPrintPodLogs(taskRunName, namespace string) error {
+	podName := getPodNameFromTaskRun(taskRunName)
+	pod := &corev1.Pod{}
+
+	// Get the details of the Pod
+	if err := p.Client.CtrlRuntimeClient().Get(context.Background(), ctrlclient.ObjectKey{Name: podName, Namespace: namespace}, pod); err != nil {
+		return err
+	}
+
+	// Iterate through each container in the Pod to fetch logs
+	for _, container := range pod.Spec.Containers {
+		logrus.Infof("Fetching logs for container '%s' in Pod '%s'", container.Name, podName)
+
+		podLogOpts := corev1.PodLogOptions{Container: container.Name}
+		if p.args.TailLines > 0 {
+			podLogOpts.TailLines = &p.args.TailLines
+		}
+
+		req := p.Client.KubeClient().CoreV1().Pods(namespace).GetLogs(podName, &podLogOpts)
+
+		podLogs, err := req.Stream(context.Background())
+		if err != nil {
+			logrus.Errorf("Failed to fetch logs for container '%s', %v", container.Name, err)
+			continue
+		}
+
+		// Read and display the logs
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, podLogs)
+		podLogs.Close()
+		if err != nil {
+			logrus.Errorf("Failed to read logs for container '%s', %v", container.Name, err)
+			continue
+		}
+
+		logrus.Infof("Logs from container '%s':\n%s", container.Name, buf.String())
+	}
+
+	return nil
+}
+
+// generatePodName gets the pod name for a taskrun. The taskRunName and the name of the pod executing the task differ only by "-pod"
+func getPodNameFromTaskRun(taskRunName string) string {
+	return taskRunName + "-pod"
```

Comment:
why suffixed by `-pod`

Reply:
the pod is created by tekton, and the pod name is exactly same as taskrun name + "-pod"

Reply:
-pod is a little bit redundant

Reply:
I can't handle the naming for pod.  what I can do is to find what is the pod name for giving taskRun name